packages/frontend-app-api/src/wiring/createApp.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createExtension, createPlugin } from '@backstage/frontend-plugin-api';
+import { createInstances } from './createApp';
+
+import { MockConfigApi } from '@backstage/test-utils';
+
+describe('createInstances', () => {
```

Comment:
Let's add test cases for immediate cycles, i.e. a -> a, as well as small, a -> b -> a, and larger ones like a -> b -> c -> d -> b. Gonna need to

Reply:
Done: https://github.com/backstage/backstage/pull/20032/commits/fdf862d99957ece3fdae73a7583f181ab8af9d17